src/main/scala/tv/teads/wiremock/extension/FreeMarkerRenderer.scala

```diff
@@ -79,4 +83,44 @@ class FreeMarkerRenderer extends ResponseDefinitionTransformer {
       hash
   }
 
+  class FindFirstInArray(requestBody: JsonNode) extends TemplateMethodModelEx {
+    import freemarker.template.TemplateModelException
+
+    @tailrec
+    private def findChildNode(parent: JsonNode, fullPath: String): JsonNode = {
```

Comment:
is it possible that parent==null and this trows a NPE?

Reply:
No, if the path of the parent does not exist, the check on the condition will be false and a TemplateModelException `Value not found .. ` will be thrown. Like mentioned in Jackson documentation, here is the return type of the `findPath`method : Value of first matching node found; or if not found, a "missing node" (non-null instance that has no value)